src/compiler/types.ts

```diff
@@ -940,7 +940,8 @@ module ts {
 
     export interface ExportAssignment extends Declaration, ModuleElement {
         isExportEquals?: boolean;
-        expression: Expression;
+        expression?: Expression;
+        type?: TypeNode;

```

Comment:
There were actually two things that I found odd:
1. The type is only allowed if isExportEquals is false
2. type and expression are mutually exclusive, yet this interface implies that both or neither can be there.

That said, if you tried it and it's too much added complexity, that's fine. We're already half... you know